### PR TITLE
fix(SR-12): Stop exposing full error messages and stack traces

### PR DESCRIPTION
**Pull request by JennyHumphries**
## Introduction
This PR is to fix the [Information exposure through a stack trace](https://github.com/UK-Export-Finance/dtfs2/security/code-scanning?query=is%3Aopen+branch%3Amain+stack+trace++) alerts

## Resolution
Where errors are returned from the api methods, instead of returning them, log them and return an error message instead.

### CODE REVIEW

File: trade-finance-manager-api/src/v1/api.js [hunk 1]
```diff
@@ -214,7 +214,8 @@ const updateDeal = async (dealId, dealUpdate) => {
 
     return response.data;
   } catch (error) {
-    return error;
```
Comment (JennyHumphries) [inline trade-finance-manager-api/src/v1/api.js:217]: There seem to be quite a few other method in this file that have
```
catch(error) {
  return error;
}
```
But that haven't been flagged by the code scanning alerts.

Should I go through and change them to follow this pattern anyway? Or are we safe to assume that the code scanning should have picked up any that we need to worry about?

Comment (ChristianMcc) [reply trade-finance-manager-api/src/v1/api.js:217]: Can you check a couple of these and just make sure that it isn't then returned by the controller that has called this api.js file please? And if it's not then we should be happy to assume the code scanning is doing it's job properly :)

File: dtfs-central-api/src/v1/controllers/cron-jobs/cron-jobs.controller.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ exports.deleteAllEstoreLogs = async (req, res) => {
 
     return res.status(200).send();
   } catch (error) {
-    return res.status(500).send(error);
```
Comment (abhi-markan) [inline dtfs-central-api/src/v1/controllers/cron-jobs/cron-jobs.controller.js:11]: Can we please add

```javascript
console.error('CRON job error %O', error);
```

File: dtfs-central-api/src/v1/controllers/durable-functions/durable-functions.controller.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ exports.deleteAllDurableFunctions = async (req, res) => {
 
     return res.status(200).send();
   } catch (error) {
-    return res.status(500).send(error);
```
Comment (abhi-markan) [inline dtfs-central-api/src/v1/controllers/durable-functions/durable-functions.controller.js:11]: Can we please add

``javascript
console.error('ACBS DOF error %O', error);
```

File: portal/server/api.js [hunk 4]
```diff
@@ -729,7 +729,7 @@ const getUnissuedFacilitiesReport = async (token) => {
     return response.data;
   } catch (error) {
     console.error('Unable to return unissued facilities', { error });
```
Comment (abhi-markan) [inline portal/server/api.js:731]: Can this please be changed to following:

```javascript
console.error('Unable to return unissued facilities %O', error);
```

File: portal/server/api.js [hunk 5]
```diff
@@ -747,7 +747,7 @@ const getUkefDecisionReport = async (token, payload) => {
     return response.data;
   } catch (error) {
     console.error('Unable to return Ukef decision report', { error });
```
Comment (abhi-markan) [inline portal/server/api.js:749]: Can this please be changed to following:

```javascript
console.error('Unable to return UKEF decision report %O', error);
```

File: trade-finance-manager-api/src/v1/api.js (hunk 1 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/api.js:217]: If you can go through them as well please that will be great, ideally if they can have the same `console.error` and return statement that will be great.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Only few minor comments to address please.

**Pull request merged**